lib/puppet_litmus/puppet_helpers.rb

```diff
@@ -147,6 +147,40 @@ def create_manifest_file(manifest)
     end
   end
 
+  # Writes a string variable to a file on a target node at a specified path.
+  #
+  # @param content [String] String data to write to the file.
+  # @param destination [String] The path on the target node to write the file.
+  # @return [Bool] Success. The file was succesfully writtne on the target.
+  def write_file(content, destination)
+    Honeycomb.start_span(name: 'litmus.write_file') do |span|
+      ENV['HTTP_X_HONEYCOMB_TRACE'] = span.to_trace_header
+      span.add_field('litmus.destination', destination)
+
+      require 'tmpdir'
+      target_node_name = ENV['TARGET_HOST']
+      tmp_filename = File.join(Dir.tmpdir, "litmus_tmp_#{Time.now.strftime('%Y%m%d')}_#{Process.pid}_#{rand(0x100000000).to_s(36)}")
```

Comment:
This looks insecure. Please use https://ruby-doc.org/stdlib-2.7.1/libdoc/tempfile/rdoc/Tempfile.html instead.